rest/route/middleware.go

```diff
@@ -485,14 +485,21 @@ func (m *TaskAuthMiddleware) ServeHTTP(rw http.ResponseWriter, r *http.Request,
 			return
 		}
 	}
-	if code, err := data.CheckTaskSecret(taskID, r); err != nil {
+	t, code, err := data.CheckTaskSecret(taskID, r)
+	if err != nil {
 		gimlet.WriteResponse(rw, gimlet.MakeJSONErrorResponder(gimlet.ErrorResponse{
 			StatusCode: code,
 			Message:    errors.Wrapf(err, "checking secret for task '%s'", taskID).Error(),
 		}))
 		return
 	}
-
+	if utility.StringSliceContains(evergreen.TaskCompletedStatuses, t.Status) {
```

Comment:
I believe the task could still be doing some things with Evergreen after end task finishes, like [updating check runs](https://github.com/evergreen-ci/evergreen/blob/63abeaa6ac75bb978685f44ee4ce9766041d2103/agent/agent.go#L1051-L1055), and I believe also teardown group.

A little hacky, but a quick fix that comes to mind is to give a leeway period where the task creds are still valid after the task finish time to let the final few things finish up.

Reply:
Teardown group is a good point. Added a leeway period of an hour 🤷